src/Concerns/CapturesState.php

```diff
@@ -407,6 +408,24 @@ public function scheduledTask(ScheduledTaskFinished|ScheduledTaskSkipped|Schedul
         $this->sensor->scheduledTask($event);
     }
 
+    public function prepareForNextRequest(): void
+    {
+        /** @var Core<RequestState> $this */
+        $this->flush();
+        memory_reset_peak_usage();
+
+        $trace = (string) Str::uuid();
+        $timestamp = $this->clock->microtime();
+
+        Compatibility::addHiddenContext('nightwatch_trace_id', $trace);
```

Comment:
We should swap this with `$this->resume()` once filtering has been merged.

Reply:
filtering has been merged